src/inventory_client/inventory_client.go

```diff
@@ -41,20 +46,66 @@ type EnabledHostData struct {
 	Host      *models.Host
 }
 
-func CreateInventoryClient(clusterId string, inventoryURL string, pullSecret string, logger *logrus.Logger) (*inventoryClient, error) {
+func CreateInventoryClient(clusterId string, inventoryURL string, pullSecret string, insecure bool, caPath string, logger *logrus.Logger) (*inventoryClient, error) {
 	clientConfig := client.Config{}
 	var err error
 	clientConfig.URL, err = url.ParseRequestURI(createUrl(inventoryURL))
 	if err != nil {
 		return nil, err
 	}
 
-	clientConfig.Transport = requestid.Transport(http.DefaultTransport)
+	var certs *x509.CertPool
+	if insecure {
+		logger.Warn("Certificate verification is turned off. This is not recommended in production environments")
+	} else {
+		certs, err = readCACertificate(caPath, logger)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	clientConfig.Transport = requestid.Transport(&http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: insecure,
```

Comment:
What is the current behavior?
By default InsecureSkipVerify is set to true or false?

Reply:
False, i.e. secure by default. Meaning it will fail with a custom CA or incorrect SAN when trying HTTPS.